src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -0,0 +1,579 @@
+                    throw new IOException();
+                }
+            }
+        }
+
+        internal Handshake Handshake
+        {
+            set
+            {
+                Debug.Assert(!(value < Handshake.None || value > Handshake.RequestToSendXOnXOff),
+                    "An invalid value was passed to Handshake");
+
+                if (value != _handshake)
+                {
+                    _handshake = value;
+                    if (Interop.Termios.TermiosReset(_handle, _baudRate, _dataBits, _stopBits, _parity, _handshake) != 0)
+                    {
+                        throw new ArgumentException();
+                    }
+                }
+            }
+        }
+
+        internal int DataBits
+        {
+            set
+            {
+                Debug.Assert(!(value < MinDataBits || value > MaxDataBits), "An invalid value was passed to DataBits");
+                if (value != _dataBits)
+                {
+                    _dataBits = value;
+                    if (Interop.Termios.TermiosReset(_handle, _baudRate, _dataBits, _stopBits, _parity, _handshake) != 0)
+                    {
+                        throw new ArgumentException();
+                    }
+                }
+            }
+        }
+
+        internal Parity Parity
+        {
+            set
+            {
+                Debug.Assert(!(value < Parity.None || value > Parity.Space), "An invalid value was passed to Parity");
+
+                if (value != _parity)
+                {
+                    _parity = value;
+                    if (Interop.Termios.TermiosReset(_handle, _baudRate, _dataBits, _stopBits, _parity, _handshake) != 0)
+                    {
+                        throw new ArgumentException();
+                    }
+                }
+            }
+        }
+
+        internal StopBits StopBits
+        {
+            set
+            {
+                Debug.Assert(!(value < StopBits.One || value > StopBits.OnePointFive), "An invalid value was passed to StopBits");
+                if (value != _stopBits)
+                {
+                    _stopBits = value;
+                    if (Interop.Termios.TermiosReset(_handle, _baudRate, _dataBits, _stopBits, _parity, _handshake) != 0)
+                    {
+                        throw new ArgumentException();
+                    }
+                }
+            }
+        }
+
+        internal bool DiscardNull
+        {
+            set
+            {
+                // Ignore.
+            }
+        }
+
+        internal byte ParityReplace
+        {
+            set
+            {
+                // Ignore.
+            }
+        }
+
+        internal void DiscardInBuffer()
+        {
+            if (_handle == null) throw new ObjectDisposedException(SR.Port_not_open);
+            // This may or may not work depending on hardware.
+            Interop.Termios.TermiosDiscard(_handle, Interop.Termios.Queue.ReceiveQueue);
```

Comment:
I see the comment but it is not clear to me if an error is possible here. If error is possible: is ok to ignore it? Similar below

Reply:
I'm not sure but since it all best effort, I think it would be better not to error here.